### PR TITLE
feat: enhance yarn workspace

### PR DESCRIPTION
**Pull request by Nutlope**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
feature

**What is the current behavior? (You can also link to an open issue here)**
no autocomplete for yarn workspace

**What is the new behavior (if this is a feature change)?**
add autocomplete for yarn workspace

**Additional info:**

* Recorded a [loom video](https://www.loom.com/share/0ca49e6ec5fc4e0f93bdf16b1c23ef1f) to show demo working
* Fulfills #274
* Accounted for corner case @QuiiBz mentioned in this [old PR](https://github.com/withfig/autocomplete/pull/294)

### CODE REVIEW

**Comment by Nutlope** [comment]
Hi @QuiiBz, got around to fixing the edge case you mentioned in #294. I ended up using generateSpec to do this. Mind reviewing this PR?

File: dev/yarn.ts [hunk 1]
```diff
@@ -91,6 +91,30 @@ const packageList: Fig.Generator = {
...
+    }
+    try {
+      const packageContent = JSON.parse(out);
+      const scripts = packageContent["scripts"];
+      if (scripts) {
+        const scriptKeys = [];
+        for (let i = 0; i < Object.keys(scripts).length; i++) {
+          scriptKeys.push({ name: Object.keys(scripts)[i] });
+        }
+        return scriptKeys;
```
Comment (QuiiBz) [inline dev/yarn.ts:111]: We can improve this with a `map`:
```suggestion
        return Object.keys(scripts).map(script => ({ name: script }))
```

File: dev/yarn.ts [hunk 2]
```diff
@@ -1012,29 +1036,65 @@ export const completionSpec: Fig.Spec = {
...
-              if (out.trim() == "") {
-                return [];
+        try {
+          const out = await executeShellCommand("cat package.json");
+
+          if (out.trim() == "") {
+            return { name: "workspaces" };
+          }
+          const packageContent = JSON.parse(out);
+          const workspaces = packageContent["workspaces"];
```
Comment (QuiiBz) [inline dev/yarn.ts:1050]: We should check if `workspaces` is not undefined to avoid the `try ... catch`

File: dev/yarn.ts [hunk 3]
```diff
@@ -1012,29 +1036,65 @@ export const completionSpec: Fig.Spec = {
...
+            }
+          }
+        } catch (e) {
+          return { name: "workspaces" };
+        }
+
+        return {
+          name: "workspace",
+          subcommands,
+        } as Fig.Spec;
```
Comment (QuiiBz) [inline dev/yarn.ts:1096]: No need to cast as `Fig.Spec`, it's inferred from the `generateSpec` function!

File: package-lock.json [hunk 4]
```diff
@@ -1,5272 +1,8 @@
 {
```
Comment (QuiiBz) [inline package-lock.json:1]: Could you reset this file? We use lockfile version 2, not 1

**Review by QuiiBz — CHANGES_REQUESTED** [review]
I tested it and it works incredibly well, congrats! I left you some comments to resolve before merging

**Comment by Nutlope** [comment]
Hi @QuiiBz, apologies for the delay! Doing a lot of onboarding stuff this week so it's busy. Thanks for the detailed feedback! I just implemented everything and it's ready for your review again.

**Review by QuiiBz — APPROVED** [review]
LGTM! Excited for this addition, it's going to be really useful!

**Pull request merged**